azure/keyvault/main.tf

```diff
@@ -1,21 +1,33 @@
 data "azurerm_client_config" "current" {}
 
+data "azurerm_resource_group" "rg" {
+  name = var.resource_group_name
+}
+
+resource "azurecaf_name" "caf_name" {
+  name          = lower("${local.org}${var.name}")
+  resource_type = "azurerm_key_vault"
+  suffixes      = [lower(local.environment)]
```

Comment:
In developer's naming convention, they have a postfix to determine if the KV is external or internal. An idea to solve this issue is to add a suffix with value E or I. The following code can give an idea of how we could solve it:

```hcl
variable "external_usage" {
  description = "(Optional) Determines if the keyvault is for internal or external usage."
  type        = bool
  default     = true
}

locals {
  internal_external_suffix = var.external_usage ? "e" : "i"
}

resource "azurecaf_name" "caf_name" {
  name          = lower("${local.org}-${var.name}")
  resource_type = "azurerm_key_vault"
  suffixes      = [local.internal_external_suffix, lower(local.environment)]
  clean_input   = true
}


variable "external_usage" {
  description = "(Optional) Enables the keyvault purge protection in case of accidental deletion. Default is false."
  type        = bool
  default     = false
}
```